engine/src/main/java/com/redhat/cloud/notifications/db/repositories/EventRepository.java

```diff
@@ -17,4 +53,111 @@ public Event create(Event event) {
+     * @param to the final date to filter the dates from.
+     * @return a list of events that comply with the provided filters.
+     */
+    public List<Event> findEventsToExport(final String orgId, final LocalDate from, final LocalDate to) {
+        this.validateFromTo(from, to);
+
+        final StringBuilder findEventsQuery = new StringBuilder();
+        findEventsQuery.append(
+            "SELECT NEW com.redhat.cloud.notifications.models.Event( " +
+                "e.id, " +
+                "e.bundleDisplayName, " +
+                "e.applicationDisplayName, " +
+                "e.eventTypeDisplayName, " +
+                "e.created) " +
+            "FROM " +
+                "Event AS e " +
+            "WHERE " +
+                "e.orgId = :orgId"
+        );
+
+        final Map<String, Object> parameters = new HashMap<>();
+        parameters.put("orgId", List.of(orgId));
+
+        if (from != null) {
+            findEventsQuery.append(
+                " AND " +
+                    "e.created >= :createdMin"
+            );
+
+            parameters.put(
+                "createdMin",
+                Date.from(from.atStartOfDay(ZoneId.systemDefault()).toInstant())
+            );
+        }
+
+        if (to != null) {
+            findEventsQuery.append(
+                " AND " +
+                    "e.created <= :createdMax"
+            );
+
+            parameters.put(
+                "createdMax",
+                Date.from(to.atTime(MAX_HOURS, MAX_MINUTES, MAX_SECONDS, MAX_NANOSECONDS).toInstant(ZoneOffset.UTC))
+            );
+        }
+
+        final TypedQuery<Event> findEventsRanged = this.entityManager.createQuery(findEventsQuery.toString(), Event.class);
+
+        for (final Map.Entry<String, Object> entry : parameters.entrySet()) {
+            findEventsRanged.setParameter(entry.getKey(), entry.getValue());
+        }
+
+        return findEventsRanged.getResultList();
+    }
+
+    /**
+     * <p>Validates that the initial and end date of the filters are valid and
+     * throws an exception if they are not. The constraints are:
+     * <ul>
+     *     <li>{@code null} dates are considered valid.</li>
+     *     <li>The initial date can't be older than a month.</li>
+     *     <li>The initial date can't be in the future.</li>
+     *     <li>If both the initial and end date are provided, the initial date
+     *     can't be after the end date.</li>
+     *     <li>The end date can't be older than a month.</li>
+     *     <li>The end date can't be in the future.</li>
+     * </ul></p>
+     *
+     * @param from the initial date of the filter.
+     * @param to the end date of the filter.
+     */
+    protected void validateFromTo(final LocalDate from, final LocalDate to) {
+        final LocalDate today = LocalDate.now();
+        final LocalDate aMonthAgo = today.minusMonths(1);
+
+        if (from != null) {
+            if (from.compareTo(today) > 0) {
```

Comment:
Would that work? The obvious method name makes the code easier to understand/maintain.
```suggestion
            if (from.isAfter(today)) {
```
If that works, the same kind of change could be applied to all `if` conditions below.